src/main/java/de/dataelementhub/model/handler/element/ElementPathHandler.java

```diff
@@ -0,0 +1,134 @@
+          .stream().filter(p -> p.getStatus() != Status.OUTDATED).collect(Collectors.toList());;
+      partialPaths.remove(partialPath);
+      if (parentScopedIdentifiers.size() > 0) {
+        for (ScopedIdentifier parentScopedIdentifier : parentScopedIdentifiers) {
+          String parentUrn = IdentificationHandler.toUrn(ctx, parentScopedIdentifier);
+          partialPaths.add(parentUrn + "/" + partialPath);
+        }
+      } else {
+        Identification namespaceIdentification = IdentificationHandler
+            .getNamespaceIdentification(ctx, urn);
+        if (!DaoUtil.accessLevelGranted(namespaceIdentification.getIdentifier(),
+            userId, DaoUtil.READ_ACCESS_TYPES)) {
+          partialPaths.remove(partialPath);
+        } else {
+          partialPaths.add(namespaceIdentification.getNamespaceUrn() + "/" + partialPath);
+        }
+      }
+    }
+    boolean pathsCompleted = partialPaths.stream().allMatch(n -> n.contains("namespace"));
+    if (!pathsCompleted) {
+      partialPaths = completePaths(ctx, userId, partialPaths);
+    }
+    return partialPaths;
+  }
+
+  /**
+   * Get the designations for all urns in a path.
+   */
+  public static List<String> getDesignations(CloseableDSLContext ctx,
+      List<String> paths, String languages) {
+    List<String> pathDesignationsList = new ArrayList<>();
+    for (String path : paths) {
+      String[] pathUrns = path.split("/");
+      String pathDesignations = "";
+      for (int i = 0; i < pathUrns.length; i++) {
+        if (i == pathUrns.length - 1) {
+          ScopedIdentifier scopedIdentifier =
+              IdentificationHandler.getScopedIdentifier(ctx, pathUrns[i]);
+          pathDesignations =
+              pathDesignations + getDesignation(ctx, scopedIdentifier.getId(), languages);
+        } else {
+          ScopedIdentifier scopedIdentifier =
+              IdentificationHandler.getScopedIdentifier(ctx, pathUrns[i]);
+          pathDesignations = pathDesignations
+              + getDesignation(ctx, scopedIdentifier.getId(), languages) + "/";
+        }
+      }
+      pathDesignationsList.add(pathDesignations);
+    }
+    return pathDesignationsList;
+  }
+
+  /**
+   * Get element designation by scopedIdentifierId.
+   */
+  public static String getDesignation(CloseableDSLContext ctx,
+      int scopedIdentifierId, String languages) {
+    Element element = new Element();
+    element.setDefinitions(DefinitionHandler.get(ctx, scopedIdentifierId));
+    element.applyLanguageFilter(languages);
+    return element.getDefinitions().get(0).getDesignation();
+  }
+
+  /**
+   * Return list of scopedIdentifiers for all parent elements.
+   */
+  public static List<ScopedIdentifier> getParentScopedIdentifiers(
+      CloseableDSLContext ctx, String urn) {
+    int scopedIdentifierId = IdentificationHandler.getScopedIdentifier(ctx, urn).getId();
+    List<ScopedIdentifier> parentElements =
```

Comment:
this variable is redundand. just directly return the ctx.select...statement

Reply:
DONE